pkg/multicast/multicast.go

```diff
@@ -374,84 +447,90 @@ func readPackageHeader(r io.Reader) (uint16, uint16, uint32, error) {
 	return n, channelID, seq, nil
 }
 
-func (c *Client) handlePackageHeader(r io.Reader) error {
-	_, channelID, seq, err := readPackageHeader(r)
+func (c *Client) handlePackageHeader(r io.Reader) (offset int, channelID uint16, seq uint32, err error) {
+	_, channelID, seq, err = readPackageHeader(r)
 
 	if err != nil {
 		c.log.Errorw("failed to decode events", "err", err)
-		return err
+		return
 	}
 
 	lastSeq, ok := c.getSeqNum(channelID)
-	if ok && seq != lastSeq+1 { // check for invalid sequence number
-		if seq == 0 {
-			return ErrConnectionReset
-		}
-		return ErrLostPackage
-	}
-	c.setSeqNum(channelID, seq)
-	return nil
+	if !ok {
+		queue := common.NewEventQueue(defaultEventQueueLength)
+		c.setEventQueue(channelID, queue)
+		return
+	} else if seq != 0 && seq <= lastSeq { // check for duplicated package
```

Comment:
Ignore the package only when:
- `seq <= lastSeq` and `lastSeq - seq < 100`
- `seq > lastSeq` and `max(uint32) - seq + lastSeq < 100`